src/app/modal/modal-close.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, HostListener } from '@angular/core';
+
+import { MzModalComponent } from './modal.component';
+
+@Directive({
+  selector: 'a[mzModalClose], button[mzModalClose], a[mz-modal-close], button[mz-modal-close]',
+})
+export class MzModalCloseDirective {
+
+  @HostListener('click') onclick() {
+    this.modalComponent.close();
+  }
+
+  constructor(
+    private modalComponent: MzModalComponent,
```

Comment:
How do you know it's the modal component where the directive is used instead of another modal component in the page. If the page contains more than one modal.